types/invariant/index.d.ts

```diff
@@ -11,8 +13,7 @@ export as namespace invariant;
 
 declare namespace invariant {
   interface InvariantStatic {
-    (testValue:false, format?:string, ...extra:any[]):never;
```

Comment:
IMO we should keep this overload to not break existing test (see comment below).